common/sessionFilters.ts

```diff
@@ -0,0 +1,162 @@
+        dateRange: 'range';
+        /** the starting date of the range */
+        startDate?: string;
+        /** the ending date of the range */
+        endDate?: string;
+    };
+
+/** Filter on calendar dates */
+export interface DateRange {
+    /** the starting date of the range */
+    startDate?: moment.Moment;
+    /** the ending date of the range */
+    endDate?: moment.Moment;
+}
+
+/** parses a query string into a DateRange (sanitizes and parses dates) */
+export function parseDateRange(query: DateRangeQuery): DateRange {
+    query = typeof query === 'object' ? query : {};
+    let startDate: moment.Moment | undefined;
+    let endDate: moment.Moment | undefined;
+    const today = moment({ hour: 0 });
+    if (typeof query.dateRange === 'string') {
+        query.dateRange = query.dateRange.trim().toLowerCase() as any;
+        switch (query.dateRange) {
+            case 'this_week':
+                startDate = today.clone().day(0);
+                endDate = today.clone().day(6);
+                break;
+            case 'today':
+                startDate = today.clone();
+                endDate = today.clone().add({ hour: 24 });
+                break;
+            case 'range':
+                if (query.startDate) startDate = moment(query.startDate);
+                if (query.endDate) endDate = moment(query.endDate);
+                break;
+        }
+    }
+    return { startDate, endDate };
+}
+
+/** Filter on times of day */
+export type TimeRangeQuery = {
+    /**
+     * undefined means no filter
+     * 'morning', 'lunch' and 'after_school' are preset time ranges
+     * 'range' is a custom time range
+     */
+    timeRange?: undefined | 'morning' | 'lunch' | 'after_school';
+} |
+    {
+        timeRange: 'range',
+        /** the start time of the time range */
+        startTime?: undefined | string,
+        /** the end time of the time range */
+        endTime?: undefined | string,
+    };
+
+/** Filter on times of day */
+export interface TimeRange {
+    /** the start time of the time range */
+    startTime?: moment.Moment;
+    /** the end time of the time range */
+    endTime?: moment.Moment;
+}
+
+const TIME_FORMAT = 'HH:mm';
+
+/** parses a time range from a query string */
+export function parseTimeRange(query: TimeRangeQuery): TimeRange {
+    query = typeof query === 'object' ? query : {};
+    let startTime: moment.Moment | undefined;
+    let endTime: moment.Moment | undefined;
+    if (typeof query.timeRange === 'string') {
+        query.timeRange = query.timeRange.trim().toLowerCase() as any;
+        switch (query.timeRange) {
+            case 'morning':
+                startTime = moment({ hour: 6 }); endTime = moment({ hour: 11 }); break;
+            case 'lunch':
+                startTime = moment({ hour: 11 }); endTime = moment({ hour: 1 }); break;
+            case 'after_school':
+                startTime = moment({ hour: 1 }); endTime = moment({ hour: 3, minute: 30 }); break;
```

Comment:
after_school should be school-specific

Reply:
I think this is more of a *could* than a *should*. I purposefully made it a pretty broad range

Reply:
And there's going to be a way to do a custom time range, anyways